index.html

```diff
@@ -850,9 +850,9 @@ <h4>Definitions</h4>
       <h3><dfn data-lt="JwsVerificationKey2020" data-dfn-type="dfn" id="JwsVerificationKey2020">JwsVerificationKey2020</dfn></h3>
 
       <p>A JWK publicKey.</p>
-
-      <p class="issue" data-number="240">
-The duplication and/or possible interaction of properties held in a JWK and a verification method are an active topic of discussion in the Working Group. Implementers are cautioned that the behavior of values associated with this property are not stable and might change in the future.
+      
+      <p class="warning">
+This key type has been deprecated in favor of JsonWebKey2020.
```

Comment:
Oof, please no. Is this a private key or a public key? Name change makes it impossible to know. -1 to this change, it sets an incredibly dangerous precendent that we specifically avoided by using "Verification" in the type.

Reply:
It also leads to a cambrian explosion of needing to register a different key for each function we plan to use it for. For example, I need to create JwsDerivationKey2020, so that I can take the exact same JWK (when using P curves) that I would use for JwsVerificationKey2020.

Reply:
it uses `publicKeyJwk`... so I find this argument not very strong...

Reply:
> For example, I need to create EcdhJwkDerivationKey2020, so that I can take the exact same JWK (when using P curves) that I would use for JwsVerificationKey2020.

Yes, that is exactly what we don't want to happen... you are proposing an anti-pattern. The anti-pattern is copy-pasting the same key material all over the place for different uses, it leads to accidental key reuse. We want to make that a painful process, because developers that don't know better quite often re-use keys for different purposes w/o understanding the ramifications.

@OR13 wrote:
> it uses publicKeyJwk... so I find this argument not very strong...

Yeah, but you could just as easily use privateKeyJwk, right? That's the difference, publicKeyJwk and privateKeyJwk are *both* legal in JsonWebKey2020. In JwkVerificationKey2020, you can only use publicKeyJwk... use of privateKeyJwk is forbidden. Forbidding the use of privateKeyJwk, and enforcing that as a part of the standard, greatly reduces the security attack surface and greatly reduces the chance that someone is going to upload a JsonWebKey2020 with a privateKeyJwk to a DLT, pwn'ing themselves and their customers in the process.

This PR is a ticking time bomb... strong -1. We should talk about this on a call somewhere, I don't think you guys have thought through the ramifications of this PR.

Reply:
Technically you can verify with a private key jwk.... so i don't agree with your logic...

The thing we want is a way to say "This is a JSON Web Key".... we are explicitly not trying to say: "This JSON Web Key is for Verification".... your argument is essentially an objection to JWK, which is defined here: https://tools.ietf.org/html/rfc7517

There is nothing stopping people from setting `kid` to their SSN, or using the fragment part of publicKey.id to encode a routing number...

similarly, people can embed private keys in did documents, if they wish, here is an example of one: https://github.com/transmute-industries/vc.transmute.world/blob/master/src/services/unlockedDIDs/did:v1:test:nym:z6MkhdmzFu659ZJ4XKj31vtEDmjvsi5yDZG5L7Caz63oP39k.json

There is nothing wrong with this for testing purposes... in fact, it extremely useful.

Since I've now shown how to embed private keys in a did document using base58, can we agree that naming the the verification method type has nothing to do with that?

This has more to do with support for encryption and signing with the same key type than it does with public or private keys... 

Essentially, you are arguing that we cannot use the same JWK for both signing and encryption in a did document... which will make integration with existing JOSE systems that do that... harder.... Its generally a best practice not to use the same key for signing and encryption, but its generally allowed, even by lib sodium:

https://libsodium.gitbook.io/doc/advanced/ed25519-curve25519#notes

did key, makes use of this conversion , specifically to support X25519KeyAgreementKey... which AFAIK is the only documented key type for a purpose other than signing... and well... its not really documented anywhere....

We don't want to create a new JSON-LD suite for every JOSE Key, Signature and Encryption Scheme... I'm pretty sure @selfissued @tplooker @dhh1128 and anyone else trying to use JOSE (all of DIDComm) does not want to create a new JSON-LD Suite for every JOSE algorithm.... 

Given the extension mechanisms available today, developers have a number of options for how to do this... it can be defined in did core, with warnings and clear language... it can be defined in DID Core extensions with warnings... it can be defined in DIF / CCG as a remote extension that everyone who wants to use JOSE can use... or a company can define it themselves.... so in short... we can't stop people from using JWKs for encryption and signing, any more than we can stop developers like myself for checking in private keys for test dids....

I suggest we define this stuff in DID Core, with the necessary warnings, because thats the best way to prevent harm here.

Reply:
While I agree that it's not "best practice" that sentiment comes from the rationale that because RSA based keys cannot be re-used we also shouldn't do so in ECC "just in case". However, there's been no academic literature that I can find that suggests this is dangerous for ECC curves which depend on a different hard problem. Also, as Orie points out this is a common pattern that is shared in the Montgomery (Curve25519) and Twisted Edwards curves (Ed25519) for many implementations including digital bazaar's [implementation of did:key](https://github.com/digitalbazaar/did-method-key-js/blob/89c0f6f4be80e5143c79cfd9961fbfc9de53c980/lib/driver.js#L104). Ursa just added it, libsodium supports it, dalek-cryptography supports it, and I haven't checked openssl.

Fun fact too, Montogomery curves (Curve25519) [can be converted to Weierstrass curves](https://en.wikipedia.org/wiki/Montgomery_curve) (secp/nistp curves), but no one has done that because it hasn't been necessary or been useful. I'm also not sure points would map properly over the finite fields. 

The current state of the art is that conservative estimates say you shouldn't do that, but no one has proven it's secure or insecure for ECC crypto. RSA that is not true. I can't remember the particular algorithms, but I do remember walking away knowing RSA was not safe to do and ECC had no literature on the topic.

So why does this matter? This will make a difference on IoT use cases (where security has been relaxed a bit to fit the constraints of compute and store when needed). If we allow this concept then it means that IoT use cases are allowed to store one key and rely on compute to perform encrypt/signing which is quite beneficial.